jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -330,7 +330,11 @@ static void afterClinitInitialization() {
 		/*[IF Sidecar18-SE-OpenJ9]*/
 		Properties props = internalGetProperties();
 		/*[IF JAVA_SPEC_VERSION >= 11]*/
+		/*[IF JAVA_SPEC_VERSION >= 18]*/
+		consoleDefaultCharset = Charset.forName(props.getProperty("native.encoding"), sun.nio.cs.US_ASCII.INSTANCE);
```

Comment:
JEP400 suggested
```
String encoding = System.getProperty("native.encoding");  // Populated on Java 18 and later
Charset cs = (encoding != null) ? Charset.forName(encoding) : Charset.defaultCharset();
```
Is `Charset.defaultCharset()` preferred than `sun.nio.cs.US_ASCII.INSTANCE`?

Reply:
ASCII is the safer choice. Charset.defaultCharset() is typically UTF-8. When the actual detected encoding isn't supported by Java, it seems unlikely that UTF-8 is going to work. Not sure much, if any encoding, is a superset of UTF8. Anything outside the ASCII range (assuming ASCII works at all), is likely going to show up as garbage.

Reply:
The advice in JEP 400 to use `Charset.defaultCharset()` seems to be directed at Java users rather than Java implementers.
I expect `native.encoding` will always refer to a legal charset name so the fallback value should be irrelevant.

Reply:
It's not that native.encoding won't be valid, but that Java doesn't support all encodings. Also they are split into encodings supported by java.base and extended encodings. The extended encodings are not available during bootstrap when this code is run. If necessary the console will be updated to use an extended encoding after bootstrap.

Reply:
I expect it's possible for an OS to be configured to use an encoding not supported by Java at all, in which case ASCII will be used. Or the user can set another console encoding via sun.stdout.encoding and sun.stderr.encoding.